extensions/target-specific/chromium/blink/fragment-tree/fragment-tree.js

```diff
@@ -0,0 +1,78 @@
+//--------------------------------------------------------------
+//
+//    MIT License
+//
+//    Copyright (c) Google LLC. All rights reserved.
+//
+//--------------------------------------------------------------
+
+"use strict";
+
+var NGPhysicalFragmentTree = undefined;
+Loader.OnLoad(function() {
+    function fragmentToConcreteType(fragment) {

```

Comment:
Hm, you mean make three extended fields, one for each subclass? And then here return the first one that's not null?
I was hoping there's a better way to declare an extended field that may have different types, but I guess not :/

Reply:
For types without a v-table, there's unfortunately no better solution. For types that have a v-table though, you could just use vcast. A [vtable cast] extended field is automatically added in the type explorer for types that are vcast-able, so you wouldn't need to add your own fields for these.

From what I can see, NGPhysicalFragment doesn't have a vtable, but NGPhysicalContainerFragment does? So maybe you can get a NGPhysicalContainerFragment using your extended field from the other PR, and then vcast down? Then again, I think I'm looking a different set of symbols from you, because I don't see the "buffer_" member on NGPhysicalContainerFragment.

Reply:
I made the container fragment non-virtual yesterday, because it made it harder to deal with (casts did not do the right thing). That was in https://crrev.com/c/1569517

I will update this pull request shortly

Reply:
Done